### PR TITLE
 Updated LHN layout in GSD Priority Mode

### LINKED ISSUE #1914: Update LHN layout in GSD Priority Mode
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___

## Expected Result:
- For the rows to be tightly stacked, and compact when in "GSD" Priority Mode. The avatars should be smaller and the distance between names/rooms should be less than they currently are in the "Most Recent" Priority Mode:

image

## Actual Result:
- Layout for GSD is the same as "Most Recent" (not compact)

image

## Action Performed:
- On Desktop, click your avatar at the top to show Settings on right side
- Click Preferences
- Select the Priority Mode drop down menu then enable GSD
- Observe LHN (left hand navigation) 

## Workaround:
N/A (not a bug, this is an improvement) 

## Platform:
All:
-  Web
-  iOS
-  Android
-  Desktop App
-  Mobile Web

**Notes/Photos/Videos:** Any additional supporting documentation
Above 👆 

Job posting: https://www.upwork.com/jobs/~0164a6dae10c0e9380

### PR DESCRIPTION
**Pull request by parasharrajat**
Please review @shawnborton.

### Fixed Issues

Fixes #1914

### Tests
1. Toggle PriorityMode to `GSD` from Profile preferences and observe the layout change on all platforms.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [ ] iOS
- [x] Android

#### Web
image

#### Mobile Web
image

#### Desktop
image

### CODE REVIEW

**Comment by shawnborton** [comment]
A couple of pieces of design guidance before I dive in:

The new size of the avatar area should be 28x28:
image

image

For multiple avatars, each one should be 18x18:
image

The border width around the bottom right avatar should be 3:
image

And notice that the font size inside of that bottom right bubble is 9 - we'll have to create a new fontSize variable for that:
image

The gap between the avatar and the grouping of the username/preview is 12:
image


The gap between the username and the message preview is 8:
image

**Comment by parasharrajat** [comment]
Thanks for the info @shawnborton. Matter of fact, I chose the same size as you mentioned. But still needs to adjust a few. Please review it when I take it out of the draft. Thanks.

**Comment by shawnborton** [comment]
Sounds good, just ping me when it's ready :)

**Comment by parasharrajat** [comment]
Updated @shawnborton.

**Comment by shawnborton** [comment]
Looking good! One small thing that I notice is that the outer border of the bottom right avatar in the avatar group looks strange:
image

I think this is because the border radius is not large enough, and thus we aren't getting a perfect circle for this outer border. I think this problem might actually already exist today. Anyways, I think if we just bump up the border radius to being something larger than just avatarSize/2, we should be good. Let me know if that makes sense!

**Comment by parasharrajat** [comment]
I think why it is like that. 
image

**Comment by shawnborton** [comment]
Ah interesting. Well that is a strange way to implement that. In the very least, that red portion should be larger.

**Review by ctkochan22 — DISMISSED** [review]

**Comment by ctkochan22** [comment]
@shawnborton Code looks ok. Are we still waiting on the change to make the red portion larger?

I'll default to you for final approval and merge

**Comment by shawnborton** [comment]
@parasharrajat let me know if you will be addressing the border issue mentioned above.

**Comment by parasharrajat** [comment]
Yes. I am working on it

**Comment by parasharrajat** [comment]
@shawnborton New border implementation.
borderWidth is 3 for both modes.
|Web | Android|
|------------|--------|
|image|Screenshot_2021-03-24-17-56-16-964_com expensify chat
|

**Comment by shawnborton** [comment]
Tested and this looks great! I know you didn't introduce this, but one thing I think we could benefit from is to remove the right padding from the wrapper that wraps the avatar + name + preview:

image

Change that to:
image


That will give us a little bit of extra room for the message previews to show. Thoughts on that?

**Comment by parasharrajat** [comment]
Yeah I like that. Should I do this only for GSD mode layout?

**Comment by shawnborton** [comment]
> Yeah I like that. Should I do this only for GSD mode layout?

I think for both modes is okay.

**Comment by parasharrajat** [comment]
@shawnborton Updated and Fixed the issue as well. Now working well.

**Review by shawnborton — APPROVED** [review]

**Comment by shawnborton** [comment]
Cool, thanks for updating. I think you just need to update the screenshots in the original comment above too.

**Comment by parasharrajat** [comment]
Screenshots updated. @shawnborton I think it can be merged now. https://github.com/Expensify/Expensify.cash/pull/2015#issuecomment-805318379

**Comment by shawnborton** [comment]
@ctkochan22 all you!

**Review by ctkochan22 — APPROVED** [review]

File: src/pages/home/sidebar/OptionRow.js [hunk 1]
```diff
@@ -107,20 +128,27 @@ const OptionRow = ({
                                     <MultipleAvatars
                                         avatarImageURLs={option.icons}
                                         optionIsFocused={optionIsFocused}
+                                        size={mode === 'compact' ? 'small' : 'default'}
+                                        styles={hovered && !optionIsFocused && {
+                                            secondAvatar: {
```
Comment (marcaaron) [inline src/pages/home/sidebar/OptionRow.js:133]: This should be in `styles.js` not right in the JSX I think?

File: src/components/MultipleAvatars.js [hunk 2]
```diff
@@ -10,58 +10,86 @@ const propTypes = {
...
+        // eslint-disable-next-line react/forbid-prop-types
+        singleAvatar: PropTypes.object,
+
+        // Style for Second Avatar on Multiple Avatars
+        // eslint-disable-next-line react/forbid-prop-types
+        secondAvatar: PropTypes.object,
+
+        // Style for avatar Container
+        // eslint-disable-next-line react/forbid-prop-types
+        emptyAvatar: PropTypes.object,
```
Comment (marcaaron) [inline src/components/MultipleAvatars.js:29]: I'm not following where we are getting these other style props from...

`styles.singleAvatar` and `styles.emptyAvatar`

I don't see where we are passing them as props.

Comment (marcaaron) [reply src/components/MultipleAvatars.js:29]: Also we should not be passing style objects as props anyway unless it is an array or single object of styles.

File: src/components/Avatar.js [hunk 3]
```diff
@@ -26,10 +31,10 @@ class Avatar extends PureComponent {
             <Image
                 ref={el => this.image = el}
                 source={{uri: this.props.source}}
-                style={[
-                    styles.avatarNormal,
-                    ...this.props.style,
-                ]}
+                style={_.union([
```
Comment (marcaaron) [inline src/components/Avatar.js:34]: Why `_.union()`?

File: src/components/MultipleAvatars.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React, {memo} from 'react';
 import PropTypes from 'prop-types';
 import {Image, Text, View} from 'react-native';
-import styles from '../styles/styles';
+import globalStyles from '../styles/styles';
```
Comment (marcaaron) [inline src/components/MultipleAvatars.js:4]: Not really into the `globalStyles` convention. Can we just keep this as `styles` and then if there are some more specific styles passed in as props give them a more descriptive name? Usually the prop styles have a specific purpose.

File: src/components/MultipleAvatars.js [hunk 5]
```diff
@@ -10,58 +10,86 @@ const propTypes = {
...
+    size: 'default',
+    styles: {},
 };
 
-const MultipleAvatars = ({avatarImageURLs, optionIsFocused}) => {
+const MultipleAvatars = ({
+    avatarImageURLs, optionIsFocused, size, styles,
+}) => {
+    const avatarContainerStyles = [
+        size === 'small' ? globalStyles.emptyAvatarSmall : globalStyles.emptyAvatar, styles.emptyAvatar,
```
Comment (marcaaron) [inline src/components/MultipleAvatars.js:44]: `styles.emptyAvatar` should go on a new line to make this more readable

File: src/components/MultipleAvatars.js [hunk 6]
```diff
@@ -10,58 +10,86 @@ const propTypes = {
...
+    ];
+    const singleAvatarStyles = [
+        size === 'small' ? globalStyles.singleAvatarSmall : globalStyles.singleAvatar, styles.singleAvatar,
```
Comment (marcaaron) [inline src/components/MultipleAvatars.js:47]: `styles.singleAvatar` should go on a new line to make this more readable

**Review by marcaaron — COMMENTED** [review]
Spotted a few things I think we should clean up here.

File: src/components/Avatar.js (hunk 3 above)
Comment (parasharrajat) [reply src/components/Avatar.js:34]: Somebody suggested me in a recent review so I am following that. In real, styles should be automatically merged and duplicate properties should be removed automatically.

Comment (parasharrajat) [reply src/components/Avatar.js:34]: Reverting it.

**Pull request merged**